server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/rule/MySQLAffectedRowsExceedLimit.java

```diff
@@ -120,14 +123,44 @@ public List<DialectType> getSupportsDialectTypes() {
         return Arrays.asList(DialectType.MYSQL, DialectType.OB_MYSQL);
     }
 
+    /**
+     * MySQL checks the count of value list. Process mode: case1: For INSERT INTO ... VALUES(...)
+     * statements, ODC checks the count of value list. case2: For INSERT INTO ... SELECT ... statements,
+     * ODC runs EXPLAIN statements to get affected rows.
+     *
+     * @param insertStatement target sql
+     * @return affected rows
+     */
+    private long getMySqlAffectedRowsByCount(Insert insertStatement) {
+
+        List<List<Expression>> insertList = insertStatement.getTableInsert().get(0)
```

Comment:
the logic here is not robustness, we should check three things:
1. InsertTable list is empty or not
2. InsertTable list is not empty and we should judge if the insertTable list is singleton
3. if the insert table list is singleton, we should get the first one of it
4. we should judge if the values is empty
5. if the values is empty we should judge whether it is select or const value

you can ref the `libs/ob-sql-parser/src/main/java/com/oceanbase/tools/sqlparser/statement/insert/InsertTable.java` to do this

Reply:
got it and the work has been completed.